include/oneapi/dpl/pstl/execution_impl.h

```diff
@@ -111,41 +79,72 @@ template <typename _ExecutionPolicy>
 using __allow_parallel =
     typename __internal::__policy_traits<typename ::std::decay<_ExecutionPolicy>::type>::__allow_parallel;
 
-template <typename _ExecutionPolicy, typename... _IteratorTypes>
-auto
-__is_vectorization_preferred(_ExecutionPolicy& __exec)
-    -> decltype(__internal::__lazy_and(__exec.__allow_vector(),
-                                       typename __internal::__is_random_access_iterator<_IteratorTypes...>::type()))
+//------------------------------------------------------------------------
+// backend selector with tags
+//------------------------------------------------------------------------
+
+struct __tbb_backend {};
+
+template <class _IsVector>
+struct __serial_tag
+{
+    using __is_vector = _IsVector;
+};
+
+template <class _IsVector>
+struct __parallel_tag
+{
+    using __is_vector = _IsVector;
+    // backend tag can be change depending on
+    // TBB availability in the environment
+    using __backend_tag = __tbb_backend;
```

Comment:
May be the member field of the class rather than type if we would like `__tbb_backend` to provide some useful information